core/driver/sources/src/test/scala/org/virtuslab/ideprobe/dependencies/SingleRunFixtureTest.scala

```diff
@@ -49,4 +49,40 @@ final class SingleRunFixtureTest extends IdeProbeFixture with WorkspaceFixture w
     assertFalse("Workspace was not removed", Files.exists(workspace))
     assertFalse("IDE home was not removed", Files.exists(ideHome))
   }
+
+  @Test
+  def removesDirectoriesEvenAfterFailureToRunIntelliJ(): Unit = {
+    val intelliJFixture = IntelliJFixture(
+      version = IntelliJVersion.Latest,
+      plugins = List()
+    ).withAfterIntelliJInstall((_, intellij) =>
+      Files.delete(intellij.root.resolve("bin").resolve("idea.sh")) //To prevent the IDE from launching.
+    )
+
+    val instancesBefore = Option(intelliJFixture.factory.paths.instances.toFile.list().toSet).getOrElse(Set.empty)
```

Comment:
It would be good to have some helper method for these 4 long chains of calls in this test.
There should be PathExtension util for listing files/directories, if not, it would be good to add. 
Also, why Option? How could `toSet` return null?

Reply:
Indeed there is a PathExtension method for that. Using it now.

The Option here was to avoid getting NPE, since `File.list` method can return null - but I misplaced the `toSet` call. Anyway, it's probably unnecessary as the directories should be created at this point.